src/terminal/input/terminalInput.hpp

```diff
@@ -13,28 +13,32 @@ Author(s):
 - Michael Niksa (MiNiksa) 30-Oct-2015
 --*/
 
-#include <functional>
-#include "../../types/inc/IInputEvent.hpp"
 #pragma once
 
+#include "../../types/inc/IInputEvent.hpp"
+
 namespace Microsoft::Console::VirtualTerminal
 {
     class TerminalInput final
     {
     public:
-        TerminalInput(_In_ std::function<void(std::deque<std::unique_ptr<IInputEvent>>&)> pfn);
+        using StringType = std::wstring;

```

Comment:
The choice of `std::wstring` is inoptimal because it'll heap-allocate on every single key press. I figured it doesn't really matter though, because we already do that and more.

Reply:
It's 7 chars due to the terminating null character, but the win32 key sequence is longer than that. (I mean it's also longer than 8 characters...)

Reply:
oh, win32 input sequences. _bah_.

Reply:
can you add an `operator==` to the global namespace for memory_buffer?

is it comparable in fmt 9??

Reply:
As far as I know it isn't. I'd probably just use `til::small_vector` instead, but it lacks an `append(some_range)` method. I think we should improve on this at some other time. It realistically doesn't make any difference in our overall application.